pkg/migrate/marketplacemigration.go

```diff
@@ -0,0 +1,424 @@
+
+	for _, policy := range authPolicies {
+		if policy == apic.Apikey {
+			credentialRequestPolicies = append(credentialRequestPolicies, provisioning.APIKeyCRD)
+		}
+		if policy == apic.Oauth {
+			credentialRequestPolicies = append(credentialRequestPolicies, []string{provisioning.OAuthPublicKeyCRD, provisioning.OAuthSecretCRD}...)
+		}
+	}
+
+	return credentialRequestPolicies, nil
+
+}
+
+func (m *MarketplaceMigration) checkCredentialRequestDefinitions(credentialRequestPolicies []string) []string {
+	// remove any crd not in the cache
+	knownCRDs := make([]string, 0)
+	for _, policy := range credentialRequestPolicies {
+		if def, err := m.cache.GetCredentialRequestDefinitionByName(policy); err == nil && def != nil {
+			knownCRDs = append(knownCRDs, policy)
+		}
+	}
+
+	return knownCRDs
+}
+
+func (m *MarketplaceMigration) registerAccessRequestDefinition(scopes map[string]string) (*mv1a.AccessRequestDefinition, error) {
+	oauthScopes := make([]string, 0)
+	for scope := range scopes {
+		oauthScopes = append(oauthScopes, scope)
+	}
+
+	callback := func(ard *mv1a.AccessRequestDefinition) (*mv1a.AccessRequestDefinition, error) {
+		return ard, nil
+	}
+
+	var ard *mv1a.AccessRequestDefinition
+	var err error
+	if len(scopes) > 0 {
+		ard, err = provisioning.NewAccessRequestBuilder(callback).
+			SetSchema(
+				provisioning.NewSchemaBuilder().
+					AddProperty(
+						provisioning.NewSchemaPropertyBuilder().
+							SetName("scopes").
+							SetLabel("Scopes").
+							IsArray().
+							AddItem(
+								provisioning.NewSchemaPropertyBuilder().
+									SetName("scope").
+									IsString().
+									SetEnumValues(oauthScopes)))).Register()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return ard, nil
+}
+
+// updateRI updates the resource, and the sub resource
+func (m *MarketplaceMigration) updateRI(ri *v1.ResourceInstance) error {
+	_, err := m.client.UpdateResourceInstance(ri)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m *MarketplaceMigration) createInstanceEndpoint(endpoints []apic.EndpointDefinition) ([]mv1a.ApiServiceInstanceSpecEndpoint, error) {
+	endPoints := make([]mv1a.ApiServiceInstanceSpecEndpoint, 0)
+	var err error
+
+	if len(endpoints) > 0 {
+		for _, endpointDef := range endpoints {
+			ep := mv1a.ApiServiceInstanceSpecEndpoint{
+				Host:     endpointDef.Host,
+				Port:     endpointDef.Port,
+				Protocol: endpointDef.Protocol,
+				Routing: mv1a.ApiServiceInstanceSpecRouting{
+					BasePath: endpointDef.BasePath,
+				},
+			}
+			endPoints = append(endPoints, ep)
+		}
+	} else {
+		m.logger.Debug("Processing API service instance with no endpoint")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return endPoints, nil
+}
+
+func (m *MarketplaceMigration) handleSvcInstance(
+	svcInstance *v1.ResourceInstance, revision *v1.ResourceInstance,
+) error {
+	logger := m.logger.
+		WithField("instance-name", svcInstance.Name).
+		WithField("revision-name", revision.Name)
```

Comment:
can we add the service-name here too?

Reply:
Will do 👍🏾

Reply:
We actually log service upstream... so I don't believe we need it here.

Reply:
We do, but you can't aggregate all logs together based on service name if not all logs log it.